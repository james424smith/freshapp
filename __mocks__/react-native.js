import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
