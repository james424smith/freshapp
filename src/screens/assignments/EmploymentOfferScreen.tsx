import React from "react";
import { ScrollView } from "react-native";
import PdfRender from "../../components/PdfRender";
import styles from "../../styles/assignments/EmploymentOfferStyles";
import { useSelector } from "react-redux";
import { IRootReducerType } from "../../redux/reducers";
import { AssignmentStackParamList } from "../../navigation/MainTabNavigator";
import { RouteProp } from "@react-navigation/native";
import { CURRENT } from "../../constants/assignmentConstants";
interface Props {
  route: RouteProp<AssignmentStackParamList, "EmploymentOffer">;
}
const EmploymentOfferScreen = (props: Props) => {
  const { route } = props;
  const assignmentType = route.params?.assignmentType;
  const currentEmploymentOffer = useSelector<
    IRootReducerType,
    string | undefined
  >(
    ({ assignmentDetails }) =>
      (assignmentDetails.assignmentDetails || [])[0]?.employmentOfferDocument
        ?.document
  );
  const nextEmploymentOffer = useSelector<IRootReducerType, string | undefined>(
    ({ assignmentDetails }) =>
      (assignmentDetails.assignmentDetails || [])[1]?.employmentOfferDocument
        ?.document
  );

  const employmentOfferDocument =
    assignmentType === CURRENT ? currentEmploymentOffer : nextEmploymentOffer;
  const pdf = employmentOfferDocument;
  return (
    <ScrollView contentContainerStyle={styles.scrollViewContainer}>
      {(pdf && <PdfRender document={pdf} resourceType={"base64"} />) ||
        undefined}
    </ScrollView>
  );
};

export default EmploymentOfferScreen;
