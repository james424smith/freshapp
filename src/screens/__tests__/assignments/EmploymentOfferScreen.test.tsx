import React from "react";
import EmploymentOfferScreen from "../../assignments/EmploymentOfferScreen";
import { render, cleanup } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

describe("EmploymentOfferScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when focus is true when employmentOfferDocument is present", () => {
    const route = {
      state: {},
      key: "EmploymentOffer-E0ArYKZZNpvIFYMFIEyaQ",
      name: "EmploymentOffer" as "EmploymentOffer",
      params: { assignmentType: "current" as "current" },
      getState: jest.fn(),
    };
    const mockedState = {
      assignmentDetails: {
        assignmentDetails: [
          {
            employmentOfferDocument: { document: "12345" },
          },
          {},
        ],
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <EmploymentOfferScreen route={route} />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when focus is true when employmentOfferDocument is present and assignmentType is next", () => {
    const route = {
      state: {},
      key: "EmploymentOffer-E0ArYKZZNpvIFYMFIEyaQ",
      name: "EmploymentOffer" as "EmploymentOffer",
      params: { assignmentType: "next" as "next" },
      getState: jest.fn(),
    };
    const mockedState = {
      assignmentDetails: {
        assignmentDetails: [
          {
            employmentOfferDocument: {},
          },
          {
            employmentOfferDocument: { document: "12345" },
          },
        ],
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <EmploymentOfferScreen route={route} />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when focus is true when employmentOfferDocument is undefined", () => {
    const route = {
      state: {},
      key: "EmploymentOffer-E0ArYKZZNpvIFYMFIEyaQ",
      name: "EmploymentOffer" as "EmploymentOffer",
      params: { assignmentType: "current" as "current" },
      getState: jest.fn(),
    };
    const mockedState = {
      assignmentDetails: {
        assignmentDetails: undefined,
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <EmploymentOfferScreen route={route} />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
