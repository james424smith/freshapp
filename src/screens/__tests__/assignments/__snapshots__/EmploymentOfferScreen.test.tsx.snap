// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmploymentOfferScreen snapshot test should render correct ui when focus is true when employmentOfferDocument is present 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <PDFView
        fadeInDuration={10}
        onError={[Function]}
        onLoad={[Function]}
        onPageChanged={[Function]}
        onScrolled={[Function]}
        resource="12345"
        resourceType="base64"
        style={
          Object {
            "flex": 1,
            "width": 750,
          }
        }
        textEncoding="utf-8"
        urlProps={Object {}}
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`EmploymentOfferScreen snapshot test should render correct ui when focus is true when employmentOfferDocument is present and assignmentType is next 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <PDFView
        fadeInDuration={10}
        onError={[Function]}
        onLoad={[Function]}
        onPageChanged={[Function]}
        onScrolled={[Function]}
        resource="12345"
        resourceType="base64"
        style={
          Object {
            "flex": 1,
            "width": 750,
          }
        }
        textEncoding="utf-8"
        urlProps={Object {}}
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`EmploymentOfferScreen snapshot test should render correct ui when focus is true when employmentOfferDocument is undefined 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
